### PR TITLE
Add ETH Earn Vault contracts

### PR DESCRIPTION
**Pull request by chuddster**
Ribbon EARN (ETH) Vault - use ETH staking yield for long options (no lending)

### CODE REVIEW

File: contracts/utils/STETHDepositHelper.sol [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * Swaps ETH -> stETH on Curve ETH-stETH pool, and deposits into stETH vault
+     */
+    function deposit(uint256 minSTETHAmount) external payable {
+        curveETHSTETHPool.exchange{value: msg.value}(
+            0,
+            1,
+            msg.value,
+            minSTETHAmount
+        );
+        uint256 balance = stETH.balanceOf(address(this));
```
Comment (Spablob) [inline contracts/utils/STETHDepositHelper.sol:42]: Perhaps a similar logic as in the lending helper deposit function makes sense here: 

to use the return value from the `exchange` vs `stETH.balanceOf(address(this))`

PS: also useful in the case some ETH enters the contract via eg. selfdestruct - although I don't see how a malicious actor would be incentivized for it

**Comment by Spablob** [comment]
It may be outside of the scope of this PR but noticed [here](https://github.com/ribbon-finance/ribbon-v2-earn/blob/cde279228e1e383620b5af8f99df512fa22d2e7a/contracts/vaults/EarnVault/RibbonEarnVault.sol#L1177) - lmk your thoughts on using safeTransfer vs transfer in this case?

File: contracts/utils/STETHDepositHelper.sol (hunk 1 above)
Comment (chuddster) [reply contracts/utils/STETHDepositHelper.sol:42]: yeah you are right, probably good practice. I forgot we did it this way - but there are no problems with eth <-> steth helper so far. lets just keep

**Comment by chuddster** [comment]
> It may be outside of the scope of this PR but noticed [here](https://github.com/ribbon-finance/ribbon-v2-earn/blob/cde279228e1e383620b5af8f99df512fa22d2e7a/contracts/vaults/EarnVault/RibbonEarnVault.sol#L1177) - lmk your thoughts on using safeTransfer vs transfer in this case?

Its generally always good to use safeTransfer over transfer. safeTransfer is good for making interactions with `other` ERC20's from a third party safe. In this case, we are interacting with RBN which is ours so it is fine. But it is still generally good to use safeTransfer

**Pull request merged**